dts/arm/raspberrypi/rp2040.dtsi

```diff
@@ -46,13 +46,22 @@
 	};
 
 	soc {
-		gpio0: gpio@40014000 {
-			compatible = "raspberrypi,pico-gpio";
+		pinctrl: pin-controller@40014000 {
+			compatible = "raspberrypi,pico-pinctrl";
 			reg = <0x40014000 DT_SIZE_K(4)>;
-			gpio-controller;
-			#gpio-cells = <2>;
-			label = "GPIO_0";
-			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			status = "okay";
+			label = "PINCTRL";
+
+			gpio0: gpio@40014000 {
+				compatible = "raspberrypi,pico-gpio";
+				reg = <0x40014000 DT_SIZE_K(4)>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				label = "GPIO_0";
+				status = "disabled";
+			};
```

Comment:
suggestion: add pinctrl driver first, so that you don't need to patch drivers or DT here.